src/domain/release-hash.ts

```diff
@@ -0,0 +1,48 @@
+import path from "node:path";
+import os from "node:os";
+import https from "node:https";
+import fs from "node:fs";
+import crypto from "node:crypto";
+
+export class ReleaseHashService {
```

Comment:
nit: the naming like "Service" makes me think it's going to talk RPC of some kind

Reply:
This is standard nomenclature in [domain-driven design](https://martinfowler.com/bliki/DomainDrivenDesign.html#:~:text=Domain%2DDriven%20Design%20is%20an,through%20a%20catalog%20of%20patterns.). A "service" is something is something that performs a domain action that doesn't really fit into an aggregate, entity, or value.

I can understand how it could give that impression though. In DDD functionality that deals with technical or infrastructure concerns usually aren't in the domain package so that helps with this type of disambiguation. I also get how this can seem a bit dogmatic, but I wanted to try using DDD here so figured I'd follow through on the usual naming patterns.